### PR TITLE
Deterministic epoch

### LINKED ISSUE #36: updateValidators() should should be "pure"
we need to be able to sync to the top of the chain (arrive at same validator set) without touching the parent chain.

`bridge.methods.lastCompleteEpoch()` in *updateValidators.js* fetches data "from the future" during sync, if reading form the parent chain.

### PR DESCRIPTION
**Pull request by sunify**
Resolves #36

### CODE REVIEW

File: src/tx/applyTx/checkEpochLength.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+const { Type } = require('parsec-lib');
+
+module.exports = (state, tx) => {
+  if (tx.type !== Type.EPOCH_LENGTH) {
+    throw new Error('epochLength tx expected');
+  }
+
+  state.epochLength = Number(tx.options.epochLength);
```
Comment (johannbarbie) [inline src/tx/applyTx/checkEpochLength.js]: don't we need to check contract here, to make sure that tx is legit?

File: index.js [hunk 2]
```diff
@@ -58,6 +59,9 @@ async function run() {
       balances: {}, // stores account balances like this { [colorIndex]: { address1: 0, ... } }
       unspent: {}, // stores unspent outputs (deposits, transfers)
       processedDeposit: 0,
+      epoch: 0,
+      lastEpochHeight: 0,
+      epochLength: null,
```
Comment (johannbarbie) [inline index.js]: `0` is treated as `null` in comparisons, why not use it?

Comment (johannbarbie) [reply index.js]: other idea: bridge contract and node can not work without at least 1 validator. couldn't initial epochLength be 1?

File: src/tx/applyTx/checkEpochLength.js (hunk 1 above)
Comment (johannbarbie) [reply src/tx/applyTx/checkEpochLength.js]: if we are checking chain here, how to know if node is syncing or not?

File: index.js (hunk 2 above)
Comment (sunify) [reply index.js]: Also, we can read events before setting up `initialState` and put value from the first event

Comment (johannbarbie) [reply index.js]: how do you read the first event that happened in a contract? wouldn't you need to know the block-height when it was deployed?

**Pull request merged**